webui/react/src/hooks/useSettings.ts

```diff
@@ -162,16 +164,23 @@ const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
       stateSettings[setting] = settings[setting];
     });
 
-    update(config.storageKey, stateSettings, true);
+    update(config.storageKey, () => stateSettings, true);
   }, [config, querySettings, state, update, shouldSkipUpdates]);
 
+  const stateEntry = useMemo(() => state.get(config.storageKey), [state, config.storageKey]);
   const settings: SettingsRecord<T> = useMemo(
     () =>
       ({
-        ...(state.get(config.storageKey) ?? {}),
+        ...(stateEntry ?? {}),
       } as SettingsRecord<T>),
-    [config, state],
+    [stateEntry],
   );
+  const prevSettings: SettingsRecord<T> | undefined = usePrevious(settings, undefined);
+  const returnedSettings: SettingsRecord<T> = useMemo(() => {
```

Comment:
I don't think this actually memoizes the component how you mean it too. The value in `usePrevious` still gets updated so this just delays the value by one state update if it's equal. You probably want to do a `useState` that only updates when the value of settings is different.